src/program.py

```diff
@@ -60,7 +60,9 @@ def isin(self, p) -> bool:
         return False
 
     def serialize(self) -> str:
-        return str(self.lid) + " " + self.inst + " " + [str(op.lid) + " " for op in self.operands]
+        return str(self.lid) + " " + self.inst + " " + \
+            ' '.join([(str(op.lid) if isinstance(op, Instruction) else str(op)) + " " \
+                      for op in self.operands ])
```

Comment:
isn't there a risk of printing garbage if the operand isn't an instruction? Like how can this guarantee that the output will be valid btor2? Wouldn't we want to sanitize a bit more and check if the param is an int if not an instruction?